### PR TITLE
Make the default value of RequireGlobalDependencies constant.

### PR DESCRIPTION
**Pull request by norahuang**
[AB#1630207](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1630207)

### CODE REVIEW

File: Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs [hunk 1]
```diff
@@ -507,6 +507,9 @@ private void ProcessExecuteArguments(Context context, ObjectLiteral obj, Process
             }
             processBuilder.AdditionalTempDirectories = ProcessOptionalPathArray(obj, m_executeAdditionalTempDirectories, strict: false, skipUndefined: true);
 
+
+            // Set the default value before processing unsafteOption in case unsafeOption doesn't exist.
```
Comment (mpysson) [inline Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs]: >unsafteOption [](start = 55, length = 13)

typo

**Comment by mpysson** [comment]
>         private void ProcessExecuteArguments(Context context, ObjectLiteral obj, ProcessBuilder processBuilder, ServicePipKind serviceKind)

Can we augment some DScript unit tests to make sure we have this correct now?

---
Refers to: Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs:344 in e97881c. [](commit_id = e97881c3bffbedccf18119a89902e8ca87a3ce35, deletion_comment = False)

**Comment by norahuang** [comment]
> private void ProcessExecuteArguments(Context context, ObjectLiteral obj, ProcessBuilder processBuilder, ServicePipKind serviceKind)
> 
> 
> Can we augment some DScript unit tests to make sure we have this correct now?
> 
> Refers to: Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs:344 in e97881c. [](commit_id = e97881c, deletion_comment = False)

Added

File: Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs [hunk 2]
```diff
@@ -507,6 +507,9 @@ private void ProcessExecuteArguments(Context context, ObjectLiteral obj, Process
             }
             processBuilder.AdditionalTempDirectories = ProcessOptionalPathArray(obj, m_executeAdditionalTempDirectories, strict: false, skipUndefined: true);
 
+
+            // Set the default value before processing unsafeOption in case unsafeOption doesn't exist.
+            processBuilder.Options |= Process.Options.RequireGlobalDependencies;
```
Comment (narasamdya) [inline Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs]: >processBuilder.Options |= Process.Options.RequireGlobalDependencies; [](start = 12, length = 68)

I'm not sure what you're trying to achieve here. Why do we enable it by default here, and then possibly disable it later?

**Comment by narasamdya** [comment]
>                 ProcessUnsafeOptions(context, processBuilder, unsafeOptions);

All unsafe option should be placed under the `ProcessUnsafeOptions` method.

---
Refers to: Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs:517 in 84e594d. [](commit_id = 84e594d538f2c083a8925dd12bc2cb4ab5e20540, deletion_comment = False)

File: Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs [hunk 3]
```diff
@@ -1220,9 +1223,9 @@ private void ProcessUnsafeOptions(Context context, ProcessBuilder processBuilder
             }
 
             // UnsafeExecuteArguments.requireGlobalDependencies
-            if (Converter.ExtractOptionalBoolean(unsafeOptionsObjLit, m_unsafeRequireGlobalDependencies) != false)
+            if (Converter.ExtractOptionalBoolean(unsafeOptionsObjLit, m_unsafeRequireGlobalDependencies) == false)
```
Comment (narasamdya) [inline Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs]: >Converter.ExtractOptionalBoolean(unsafeOptionsObjLit, m_unsafeRequireGlobalDependencies) == false [](start = 16, length = 97)

Why can't we say 
```
Converter.ExtractOptionalBoolean(unsafeOptionsObjLit, m_unsafeRequireGlobalDependencies) == true
```
?

**Comment by narasamdya** [comment]
Do we need to change (almost) all files in `Public/Sdk/UnitTests/Transformers/Test.execute` ?

File: Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs (hunk 2 above)
Comment (norahuang) [reply Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.Process.cs]: We are trying to make the feature opt-in. So the logic is enable it first. Later on if it is specified in the unsafe option to opt it out, we disable it.

**Pull request merged**